app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterView.kt

```diff
@@ -7,47 +7,45 @@ import com.puzzlebench.clean_marvel_kotlin.domain.model.Character
 import com.puzzlebench.clean_marvel_kotlin.presentation.MainActivity
 import com.puzzlebench.clean_marvel_kotlin.presentation.adapter.CharacterAdapter
 import com.puzzlebench.clean_marvel_kotlin.presentation.extension.showToast
+import com.puzzlebench.clean_marvel_kotlin.presentation.fragment.DetailCharacterFragmentDialog
 import kotlinx.android.synthetic.main.activity_main.*
 import java.lang.ref.WeakReference
 
-class CharecterView(activity: MainActivity) {
+class CharacterView(activity: MainActivity) {
     private val activityRef = WeakReference(activity)
```

Comment:
You don't need to hold a weak reference to the activity. If you subscribe/unsubscribe the presenter from the activity/fragment you can assume the activity never will be null. 
Please mark the activity as private val and use it instead this weak reference in this class. So, you won't need to do this anymore: `activityRef.get()?`